docs/root/start/sandboxes/golang-network.rst

```diff
@@ -0,0 +1,83 @@
+.. _install_sandboxes_golang_network:
+
+Golang network filter
+=====================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`netcat <start_sandboxes_setup_netcat>`
+        Used to send ``TCP`` data.
+
+In this example, we show how the `Golang <https://go.dev/>`_ network filter can be used with the Envoy
+proxy.
+
+The Go plugin proxies TCP network connections. Proxied responses are prefixed with "hello, ".
+
+It also shows how Go plugins can be independently configured at runtime.
+
+Step 1: Compile the go plugin library
+*************************************
+
+Change to the ``examples/golang-network`` directory and build the go plugin library.
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/golang-network
+   $ docker compose -f docker-compose-go.yaml run --rm go_plugin_compile
+
+The compiled library should now be in the ``lib`` folder.
+
+.. code-block:: console
+
+   $ ls lib
+   simple.so
+
+Step 2: Start all of our containers
+***********************************
+
+Start all the containers.
+
+.. code-block:: console
+
+  $ docker compose pull
+  $ docker compose up --build -d
+  $ docker compose ps
+
+   NAME                            COMMAND                  SERVICE             STATUS              PORTS
+   golang-network-echo_service-1   "/tcp-echo"              echo_service        running
+   golang-network-proxy-1          "/docker-entrypoint.…"   proxy               running             0.0.0.0:10000->10000/tcp
+
+In this example, we start up two containers - an echo service which simply responds what it received from its tcp connections, and a proxy service that utilizes a golang plugin to forward data to the echo service.
+
+The destination to which the golang plugin proxies data is specified by custom configuration.
+
+.. code-block:: yaml
+   :emphasize-lines: 4
+
+   plugin_config:
+      "@type": type.googleapis.com/xds.type.v3.TypedStruct
+      value:
+         echo_server_addr: echo_service
```

Comment:
i think move all of this information into the intro

Reply:
add more info into the intro

Reply:
i think move this and consolidate the intro - this isnt really relevant to "starting the containers" and is useful as a description of how the sandbox works

Reply:
done, also move the yaml code to the intro, and simply describe how the example works.